### PR TITLE
Char sequence `\\'` in Escaped String Literals result in ParsingException

### PR DESCRIPTION
**Pull request by smakalias**
## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by smakalias** [comment]
@seut gave it another try: here I considered handling the `\\'` case when we process the input string but had to change the way we handle the normal escaped char .i.e `''` which I don't like that much and I believe it is an issue at sql formatter. 

Have a look, I'll try again the parser alternative

**Comment by seut** [comment]
@smakalias lgtm, but we then must not convert `''` into `'` and via versa nowhere for escaped literals (don't use `unquote` or `quote` for escaped literals). I think it is even more correct to handle all escaping (`'` + `\`) in one place.

**Comment by smakalias** [comment]
@seut ok, is it good to go then? Please ack..

File: sql-parser/src/main/java/io/crate/sql/Literals.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public static String quoteStringLiteral(String literal) {
     }
 
     public static String escapeAndQuoteStringLiteral(String literal) {
```
Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/Literals.java]: maybe rename to `quoteEscapedStringLiteral` ?

File: sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 2]
```diff
@@ -780,13 +780,19 @@ public void testSelectStmtBuilder() throws Exception {
         printStatement("select * from t where source_column ~ pattern_column");
         printStatement("select * from t where ? !~ ?");
         // escaped chars related
-        printStatement("select * from t where a like E'aValue'");
-        printStatement("select * from t where a = E'\\141Value'");
-        printStatement("select * from t where a = e'\\141Value'");
-        printStatement("select * from t where a = e'aa\\'bb'");
-        printStatement("select e.a from t e where e.a = E'\\141Value'");
+        printStatement("select E'aValue'");
```
Comment (seut) [inline sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java]: maybe move escaped literal test cases to a dedicated test

**Pull request merged**